pkg/metronome/client.go

```diff
@@ -79,6 +83,57 @@ func (c *Client) Jobs(opts ...JobsOption) ([]Job, error) {
 	return jobs, err
 }
 
+func (c *Client) addOrUpdate(filename string, add bool) (*Job, error) {
+	jsonFile, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	jsonBytes, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var job Job
+	if err = json.Unmarshal(jsonBytes, &job); err != nil {
+		return nil, err
+	}
+
+	var req *http.Request
+	buf := bytes.NewBuffer(jsonBytes)
+	if add {
+		req, err = c.http.NewRequest("POST", "/v1/jobs", buf, httpclient.FailOnErrStatus(true))
+	} else {
+		req, err = c.http.NewRequest("PUT", "/v1/jobs/"+job.ID, buf, httpclient.FailOnErrStatus(true))
+	}
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if err = json.NewDecoder(resp.Body).Decode(&job); err != nil {
+		return nil, err
+	}
+
+	return &job, nil
+}
+
+// Add creates a job
+func (c *Client) Add(filename string) (*Job, error) {
```

Comment:
I'd rather make this taking a `*Job` and parsing the filename in the command, to have a clear separation between the CLI and the client.

Reply:
I think this should take a Job struct directly as argument, and the caller is responsible for parsing the stream. It'd act as a basic sanity check, at least when there is something obviously broken in the JSON we don't have to bother with calling the server just to get a parse error in return. It'd also make the package API more readable for users, as having well-defined arguments is easier to understand than simply streams.